src/CodeGen_OpenGLCompute_Dev.cpp

```diff
@@ -0,0 +1,382 @@
+        }
+    } else {
+        if (type.is_float()) {
+            // no prefix for float vectors
+        } else if (type.is_bool()) {
+            oss << "b";
+        } else if (type.is_int()) {
+            oss << "i";
+        } else if (type.is_uint()) {
+            oss << "u";
+        } else {
+            internal_error << "GLSL: invalid type '" << type << "' encountered.\n";
+        }
+        oss << "vec" << type.width;
+    }
+    return oss.str();
+}
+
+// string CodeGen_OpenGLCompute_Dev::CodeGen_OpenGLCompute_C::print_reinterpret(Type type, Expr e) {
+//     ostringstream oss;
+//     oss << "as_" << print_type(type) << "(" << print_expr(e) << ")";
+//     return oss.str();
+// }
+
+namespace {
+string simt_intrinsic(const string &name) {
+    if (ends_with(name, ".__thread_id_x")) {
+        return "gl_LocalInvocationID.x";
+    } else if (ends_with(name, ".__thread_id_y")) {
+        return "gl_LocalInvocationID.y";
+    } else if (ends_with(name, ".__thread_id_z")) {
+        return "gl_LocalInvocationID.z";
+    } else if (ends_with(name, ".__thread_id_w")) {
+        internal_error << "4-dimension loops with " << name << " are not supported\n";
+    } else if (ends_with(name, ".__block_id_x")) {
+        return "gl_WorkGroupID.x";
+    } else if (ends_with(name, ".__block_id_y")) {
+        return "gl_WorkGroupID.y";
+    } else if (ends_with(name, ".__block_id_z")) {
+        return "gl_WorkGroupID.z";
+    } else if (ends_with(name, ".__block_id_w")) {
+        internal_error << "4-dimension loops with " << name << " are not supported\n";
+    }
+    internal_error << "simt_intrinsic called on bad variable name: " << name << "\n";
+    return "";
+}
+
+int thread_loop_workgroup_index(const string &name) {
+    string ids[] = {".__thread_id_x",
+                    ".__thread_id_y",
+                    ".__thread_id_z",
+                    ".__thread_id_w"};
+     for (size_t i = 0; i < sizeof(ids) / sizeof(string); i++) {
+        if (ends_with(name, ids[i])) { return i; }
+     }
+     return -1;
+}
+}
+
+void CodeGen_OpenGLCompute_Dev::CodeGen_OpenGLCompute_C::visit(const Div *op) {
+    visit_binop(op->type, op->a, op->b, "/");
+}
+
+void CodeGen_OpenGLCompute_Dev::CodeGen_OpenGLCompute_C::visit(const Mod *op) {
+    visit_binop(op->type, op->a, op->b, "%");
+}
+
+void CodeGen_OpenGLCompute_Dev::CodeGen_OpenGLCompute_C::visit(const Cast *op) {
+    Type value_type = op->value.type();
+    // If both types are represented by the same GLSL type, no explicit cast
+    // is necessary.
+    if (map_type(op->type) == map_type(value_type)) {
+        Expr value = op->value;
+        if (value_type.code == Type::Float) {
+            // float->int conversions may need explicit truncation if the
+            // integer types is embedded into floats.  (Note: overflows are
+            // considered undefined behavior, so we do nothing about values
+            // that are out of range of the target type.)
```

Comment:
This code originated from OpenGL codegen.
Since OpenGL Compute only supports floats and uin32_t for now, I removed this code as well the code that maps small ints to float in map_type() above.